forc-plugins/forc-client/src/util/gas.rs

```diff
@@ -34,18 +40,73 @@ pub(crate) async fn get_gas_used(mut tx: Script, provider: &Provider) -> Result<
         // and increase the witness limit
         tx.witnesses_mut().push(Default::default());
     }
+    let consensus_params = provider.consensus_parameters();
 
     // Get `max_gas` used by everything except the script execution. Add `1` because of rounding.
-    let consensus_params = provider.consensus_parameters();
     let max_gas_per_tx = consensus_params.tx_params().max_gas_per_tx();
     let max_gas = tx.max_gas(consensus_params.gas_costs(), consensus_params.fee_params()) + 1;
     // Increase `script_gas_limit` to the maximum allowed value.
     tx.set_script_gas_limit(max_gas_per_tx - max_gas);
 
-    let tolerance = 0.1;
-    let gas_used = provider
-        .dry_run_and_get_used_gas(tx.clone().into(), tolerance)
-        .await?;
+    get_gas_used(Transaction::Script(tx), provider).await
+}
 
+/// Returns gas_used for an arbitrary tx, by doing dry run with the provided `Provider`.
+pub(crate) async fn get_gas_used(tx: Transaction, provider: &Provider) -> Result<u64> {
+    let tolerance = 0.1;
+    let gas_used = provider.dry_run_and_get_used_gas(tx, tolerance).await?;
```

Comment:
I initially wanted to use this but SDK either does not support using this routine with Create transactions or they have a bug.

Reply:
Is there a tracking issue with the SDK for this? It would be good to add a TODO comment with a link to the issue.

Reply:
There is none, because I just discovered this while working on this. I will open an issue shortly